src/main/java/org/hibernate/models/internal/dynamic/DynamicAnnotationUsage.java

```diff
@@ -109,4 +118,46 @@ private static <A extends Annotation> Map<String, Object> extractBaselineValues(
 		}
 		return values;
 	}
+
+	private static <A extends Annotation> Map<String, Object> extractBaselineValues(
+			AnnotationDescriptor<A> annotationDescriptor,
+			AnnotationTarget target,
+			SourceModelBuildingContext context) {
+		final HashMap<String, Object> values = new HashMap<>();
+		for ( AttributeDescriptor<?> attribute : annotationDescriptor.getAttributes() ) {
+			final Object defaultValue = attribute.getAttributeMethod().getDefaultValue();
+			if ( defaultValue instanceof Annotation annotation ) {
+				try {
+					values.put( attribute.getName(), extractDynamicAnnotationUsage( annotation, target, context ) );
+				}
+				catch (InvocationTargetException | IllegalAccessException e) {
+					throw new AnnotationAccessException( "Error accessing default annotation attribute value", e );
+				}
+			}
+			else if ( attribute.isMultiValued() ) {
+				values.put( attribute.getName(), Arrays.asList( defaultValue ) );
```

Comment:
I think `Arrays.asList( defaultValue )` will produce a list containing the array itself. It should rather be:
```suggestion
				values.put( attribute.getName(), Arrays.asList( (Object[]) defaultValue ) );
```